src/Beacon.jsx

```diff
@@ -157,6 +164,23 @@ export class Beacon extends Component {
         this.setState({ targetElement, targetBounds }); // eslint-disable-line react/no-did-update-set-state
       }
     }
+
+    if (this.state.tooltip && !this.state.arrowPosition) {
+      // Calculate relative arrow position when ref is available
+      setTimeout(() => {
+        const arrowStyleLeft = this.getArrowHorizontalPosition(
+          targetElement.getBoundingClientRect(),
+          this.refs['tooltip'].getBoundingClientRect(),
```

Comment:
Our official company `.eslintrc` requires use of properties in this case rather than strings, so the thing we don't even use will just have to suffer. Interesting about the callback refs though. I didn't realize they were planning to deprecate string refs.